src/Controller/OmniaController.php

```diff
@@ -141,282 +268,218 @@ public function videoList() {
     }
 
     // Create video.
     else {
-      $media = $this->mediaEntity();
+      $media = $mediaStorage->create(['bundle' => $this->getVideoBundle()]);
       $this->logger->info('Created video "@title" (Drupal id: @id)',
         [
-          '@title' => $videoData->itemData->title,
-          '@id' => $media->id(),
+          '@title' => (string) $this->videoData->itemData->general->title,
+          '@id' => $media->uuid(),
         ]
       );
     }
 
-    $this->mapData($media, $videoData);
-    $this->setState($media, $videoData);
+    $this->mapData($media);
+    $this->setState($media);
 
     $media->save();
     $response->setData([
-      'refnr' => $videoData->itemID,
+      'refnr' => $this->getVideoId(),
       'value' => $media->id(),
     ]);
     return $response;
   }
 
   /**
-   * Get the media entity.
-   *
-   * @param int|null $id
-   *   Media entity id or NULL to create new one.
-   *
-   * @return \Drupal\Core\Entity\EntityInterface|\Drupal\media_entity\MediaInterface|null
-   *   Media entity or NULL if no matching entity is found.
-   */
-  protected function mediaEntity($id = NULL) {
-    if (!isset($this->mediaEntity)) {
-      $storage = $this->mediaEntityStorage();
-      if ($id) {
-        $this->mediaEntity = $storage->load($id);
-      }
-      else {
-        $videoBundle = $this->config('nexx_integration.settings')
-          ->get('video_bundle');
-        $this->mediaEntity = $storage->create(['bundle' => $videoBundle]);
-      }
-    }
-    return $this->mediaEntity;
-  }
-
-  /**
-   * Set proper publish state to media entity.
-   *
-   * @param \Drupal\media_entity\MediaInterface $media
-   *   Media entity.
-   * @param object $videoData
-   *   Incoming VideoData object from Nexx.tv.
+   * Prepare data from Nexx APi call.
    */
-  protected function setState(MediaInterface $media, $videoData) {
-
-    $status = Media::NOT_PUBLISHED;
-
-    if ($videoData->itemStates->active == 1
-      && $videoData->itemStates->isSSC == 1
-    ) {
-      $status = Media::PUBLISHED;
-
-      $this->logger->info('Published video "@title" (Drupal id: @id)',
-        [
-          '@title' => $videoData->itemData->title,
-          '@id' => $media->id(),
-        ]
-      );
-    }
-    else {
-      $this->logger->info('Unpublished video "@title" (Drupal id: @id)... States: 
-      active:@active 
-      isSSC:@isSSC 
-      validfrom_ssc:@validfrom_ssc 
-      validto_ssc:@validto_ssc ',
-        [
-          '@title' => $videoData->itemData->title,
-          '@id' => $media->id(),
-          '@active' => $videoData->itemStates->active,
-          '@isSSC' => $videoData->itemStates->isSSC,
-          '@validfrom_ssc' => $videoData->itemStates->validfrom_ssc,
-          '@validto_ssc' => $videoData->itemStates->validto_ssc,
-        ]
-      );
-    }
-
-    $media->set("status", $status);
+  protected function prepareData() {
+    return $this->nexxVideoData = [
+      'item_id' => $this->getVideoId(),
```

Comment:
Add checks here unless 100% sure the data and properties are there.

Reply:
I am with @mxhaut on the whole emplt() check thing. Even if yopu are sure, that those values are set, it still can change in the future. Just using the simple empty() check, as it was used before, woul be enough.

Reply:
Empty added.